src/openfermion/ops/_binary_operator.py

```diff
@@ -0,0 +1,494 @@
+        """
+
+        self.terms = []
+
+        # long string input
+        if isinstance(term, str) and '+' in term:
+            self._long_string_init(term)
+            return
+
+        # Zero operator: leave the terms list empty
+        if term is None:
+            return
+
+        # Sequence input
+        elif isinstance(term, tuple) or isinstance(term, list):  # list of tuples of tuples
+            self._parse_sequence(term)
+
+        # String input
+        elif isinstance(term, str):
+            self.terms.append(tuple(self._parse_string(term)))
+
+        # Invalid input type
+        else:
+            raise ValueError('term specified incorrectly.')
+
+        self._check_terms()
+
+    def _check_terms(self):
+        """
+        Ensures all terms obey binary rules
+        Updates terms in place.
+
+        """
+        sorted_input = []
+        for item in self.terms:
+            if len(item) == 0:
+                continue
+            binary_sum_rule(sorted_input, tuple(sorted(set(item))))
+
+        self.terms = sorted_input
+
+    def _long_string_init(self, term):
+        """
+        Initialization from a long string representation i.e.: 1 + w1 w2 + w3 w4'.
+        updates terms in place.
+
+        Args:
+            term: (str) a string representation of a symbolicBinary that involves summation
+
+        """
+        for summand in term.split(' + '):
+            # for each sum term
+            parsed_summand = self._parse_string(summand)
+            self.terms.append(parsed_summand)
+
+        self._check_terms()
+
+    def _check_factor(self, term, factor):
+        """
+        makes sure all factor in a term makes sense, cleans it up if there is multiplication with 1.
+        Args:
+            term: a single term of symbolicBinary
+            factor: a factor in term
+
+        Returns: updated term
+
+        Raises:
+            ValueError: invalid action/negative or non-integer qubit index
+
+        """
+        if len(factor) != 2:
+            raise ValueError('Invalid factor {}.'.format(factor))
+
+        term = list(term)
+        index, action = factor
+
+        if action not in self.actions:
+            raise ValueError('Invalid action in factor {}. '
+                             'Valid actions are: {}'.format(factor, self.actions))
+
+        if not isinstance(index, int) or index < 0:
+            raise ValueError('Invalid index in factor {}.'
+                             'The index should be a non-negative '
+                             'integer.'.format(factor))
+
+        if action == '1' and len(term) > 1:
+            term.remove(factor)  # no need to keep 1 multiplier in the term
+
+        return term
+
+    def _parse_sequence(self, term):
+        """
+        Parse a term given as a sequence type (i.e., list, tuple, etc.).
+        e.g. [((1,'W'),(2,'W')),...]
+
+        Args:
+            term: (list) of tuples of tuples.
+
+        """
+        if not term:
```

Comment:
Code is generally easier to read if try to avoid negatives in conditionals. For instance if you write "if term" and then what comes instead of "if not term"...